src/test/java/org/cactoos/http/HtTimedWireTest.java

```diff
@@ -76,24 +84,131 @@ public void worksFine() throws Exception {
+    @Test(timeout = 1000)
     public void failsAfterTimeout() throws Exception {
         // @checkstyle MagicNumberCheck (1 line)
         final long timeout = 100;
-        try (Socket blocker = new Socket()) {
-            blocker.connect(this.server.getLocalSocketAddress());
-            new HtTimedWire(
-                new HtWire(
-                    this.server.getInetAddress().getHostAddress(),
-                    this.server.getLocalPort()
-                ),
-                timeout
-            ).send(new InputOf("unused"));
+        final Wire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(timeout + 1);
+                return input;
+            },
+            timeout
+        );
+        new Assertion<>(
+            "fails with timeout exception",
+            () -> wire.send(new InputOf()),
+            new TimeoutExceptionMatcher()
+        ).affirm();
+    }
+
+    // @checkstyle MagicNumberCheck (1 line)
+    @Test(timeout = 1000)
+    @SuppressWarnings("unchecked")
+    // @todo #87:30m In the creation of IterableOf without
+    //  @SuppressWarnings("unchecked") complication failed with:
+    //  Warning:(124, 29) java: unchecked generic array creation
+    //  for varargs parameter of type
+    //  org.hamcrest.Matcher<? super org.cactoos.Scalar<org.cactoos.Input>>[].
+    //  Needs to investigate and fix this warning.
+    public void failsAfterTimeoutCheckIfWait() throws Exception {
+        final long timeout = 100;
+        final long sleep = 10 * timeout;
+        final Wire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(sleep);
+                return input;
+            },
+            timeout
+        );
+        final long start = System.currentTimeMillis();
+        final JoinedText message = new JoinedText(
+            "or",
+            " fails with timeout exception ",
+            " execution isn't interrupted when a timeout is exceeded "
+        );
+        new Assertion<>(
+            message.asString(),
+            () -> wire.send(new InputOf()),
+            new AllOf<>(
+                new IterableOf<>(
+                    new TimeoutExceptionMatcher(),
+                    new MaximumTimeMatcher(start, sleep)
+                ))).affirm();
+    }
+
+    /**
+     * Class that matches, that execution in not more than
+     * {@link org.cactoos.http.HtTimedWireTest.MaximumTimeMatcher#sleep}.
+     * @todo #87:30m Need to replace this with
+     *  https://github.com/llorllale/cactoos-matchers/issues/133
+     *  when an issue will be resolved. This is a temporary solution
+     *  for asserting a maximum time.
+     */
+    private final class MaximumTimeMatcher
+        extends TypeSafeDiagnosingMatcher<Scalar<Input>> {
+        private final long start;
+        private final long sleep;
+
+        MaximumTimeMatcher(final long start, final long sleep) {
```

Comment:
@aivinog1 `sleep` would best be renamed as `duration`

Reply:
Okay